lib/assert-result-factory.js

```diff
@@ -0,0 +1,33 @@
+/** @babel */
+
+class AssertResultFactory {
```

Comment:
Too much factory :p

This doesn't need to be a class. Make it a normal function instead.